src/Build.UnitTests/BackEnd/SdkResolverService_Tests.cs

```diff
@@ -85,16 +85,14 @@ public void AssertResolutionWarnsIfResolvedVersionIsDifferentFromReferencedVersi
         }
 
         [Fact]
-        public void AssertErrorLoggedWhenResolverThrows()
+        public void AssertResolverThrows()
         {
             SdkResolverService.Instance.InitializeForTests(new MockLoaderStrategy(includeErrorResolver: true));
 
             SdkReference sdk = new SdkReference("1sdkName", "version1", "minimumVersion");
 
-            var result = SdkResolverService.Instance.ResolveSdk(BuildEventContext.InvalidSubmissionId, sdk, _loggingContext, new MockElementLocation("file"), "sln", "projectPath", interactive: false, isRunningInVisualStudio: false);
-
-            result.Path.ShouldBe("resolverpath1");
-            _logger.Warnings.Select(i => i.Message).ShouldBe(new [] { "The SDK resolver \"MockSdkResolverThrows\" failed to run. EXMESSAGE" });
+            // When an SDK resolver throws, the expander will catch it and stop the build.
+            Should.Throw<SdkResolverException>(() => SdkResolverService.Instance.ResolveSdk(BuildEventContext.InvalidSubmissionId, sdk, _loggingContext, new MockElementLocation("file"), "sln", "projectPath", interactive: false, isRunningInVisualStudio: false));
```

Comment:
Assert `MockSdkResolverThrows` and `EXMESSAGE` still?

Reply:
No warnings or errors are logged in this codepath anymore, and there are other tests that verify warnings/errors.

Reply:
I was asking (confusingly) if we should verify that the exception's data fields were populated correctly.

Reply:
Added checks to ensure `MockSdkResolverThrows` and `EXMESSAGE` are contained in the exception's message.